src/luaconntrack.c

```diff
@@ -63,12 +63,14 @@ struct nf_conn *nflua_findconnid(lua_State *L)
 		derr = in6_pton(daddr, dlen, (u8 *)tuple.dst.u3.all, -1, &end);
 		break;
 	default:
-		return (struct nf_conn *)luaL_error(L, "unknown family");
+		luaL_error(L, "unknown family");
+		return NULL;
```

Comment:
`luaL_error()` never returns (it calls `longjmp()`, instead).. `return luaL_error()` is just idiomatic.. it's misleading to `return NULL` here, as this code will never be reached..

just my two satoshis.. please, feel free to ignore ;-)

Reply:
It does make sense, unfortunately this kind of usage trips compiler warnings.

It might be best for `luaL_error` to be annotated with the `noreturn` attribute in Lunatik, but at least for now, since this is the only place we've run into with these kinds of problematic casts, we figured we're happy with a simple targeted fix here instead.